### PR TITLE
refactor(GUI): remove analytics angular dependency

### PR DESCRIPTION
**Pull request by Shou**
We rename `AnalyticsService` to `analytics` and remove its dependency on
Angular.

Changelog-Entry: Rename and de-angularise AnalyticsService to analytics

### CODE REVIEW

**Comment by Shou** [comment]
What do you guys think of this one?

**Comment by lurch** [comment]
From a quick glance, looks fine to me and all the code-changes seem fairly self-explanatory. Does this tie up to @stefan-mihaila 's stuff? I guess either him or @jviotti should approve ;-)

File: lib/shared/analytics.js [hunk 1]
```diff
@@ -73,11 +49,11 @@ analytics.service('AnalyticsService', function() {
    * @param {Object} [data] - event data
    *
    * @example
-   * AnalyticsService.logEvent('Select image', {
+   * analytics.logEvent('Select image', {
    *   image: '/dev/disk2'
    * });
    */
-  this.logEvent = resinCorvus.logEvent;
+  logEvent: resinCorvus.logEvent,
```
Comment (jviotti) [inline lib/shared/analytics.js]: Can we export these functions directly? E.g:

```js
exports.logEvent = resinCorvus.logEvent;
```

File: lib/shared/analytics.js [hunk 2]
```diff
@@ -90,10 +66,30 @@ analytics.service('AnalyticsService', function() {
...
-  this.logException = resinCorvus.logException;
+  logException: resinCorvus.logException
+
+};
+
+/**
+ * @returns {Object} analytics module
+ */
+const runOnce = () => {
+  resinCorvus.install({
```
Comment (jviotti) [inline lib/shared/analytics.js]: Let's move this to the top of the file. By the way CommonJS works, this file should only be actually interpreted once. The interpreted result is then cached and returned every time the file is required.

**Comment by jviotti** [comment]
Awesome stuff @Shou , thanks a lot for handling this! Just a minor request: can we keep `analytics.js` to `lib/gui`, until we manage the move the settings model to `shared`?

**Comment by Shou** [comment]
This should be good to go now.

File: lib/gui/modules/analytics.js [hunk 3]
```diff
@@ -16,84 +16,68 @@
...
+ *
+ * @description
+ * This function sends the debug message to error reporting services.
+ *
+ * @param {String} message - message
+ *
+ * @example
+ * analytics.log('Hello World');
+ */
+module.exports.logDebug = resinCorvus.logDebug;
```
Comment (jviotti) [inline lib/gui/modules/analytics.js]: Why not just `exports.logDebug`?

Comment (Shou) [inline lib/gui/modules/analytics.js]: Done!

**Comment by Shou** [comment]
Forgot to remove an `AnalyticsService` injection from the tests. That's gone now, and the tests should pass again.

**Pull request merged**